### PR TITLE
✨ Support memory evict strategy

### LINKED ISSUE #54: [feature request] add memory evict strategy in koordlet
**What would you like to be added**:

**Why is this needed**:

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### Ⅰ. Describe what this PR does
Close #54

### CODE REVIEW

File: pkg/koordlet/resmanager/resmanager.go [hunk 1]
```diff
@@ -259,3 +271,52 @@ func (r *resmanager) hasSynced() bool {
 
 	return r.nodeSLO != nil && r.nodeSLO.Spec.ResourceUsedThresholdWithBE != nil
 }
+
```
Comment (zwzhang0107) [inline pkg/koordlet/resmanager/resmanager.go:274]: better to move evictPod killContainers to util pkg.

File: pkg/koordlet/resmanager/memory_evict.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		lastEvictTime: time.Now(),
+	}
+}
+
+func (m *MemoryEvictor) memoryEvict() {
+	klog.Infof("starting memory evict process")
+	defer klog.Infof("memory evict process completed")
+
+	if time.Now().Before(m.lastEvictTime.Add(time.Duration(m.resManager.config.MemoryEvictCoolTimeSeconds) * time.Second)) {
+		klog.Warningf("skip memory evict process, still in evict cooling time")
```
Comment (hormes) [inline pkg/koordlet/resmanager/memory_evict.go:58]: Info

File: pkg/koordlet/resmanager/memory_evict.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		return
+	}
+
+	nodeSLO := m.resManager.getNodeSLOCopy()
+	if disabled, err := isFeatureDisabled(nodeSLO, features.BEMemoryEvict); err != nil {
+		klog.Errorf("memory evict failed, error: %v", err)
```
Comment (hormes) [inline pkg/koordlet/resmanager/memory_evict.go:64]: Failed to acquire memory eviction feature-gate, error: %v ?

File: pkg/koordlet/resmanager/memory_evict.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+	memoryCapacity := node.Status.Capacity.Memory().Value()
+	if memoryCapacity <= 0 {
+		klog.Warningf("skip memory evict, memory capacity(%v) should greater than 0", memoryCapacity)
+		return
+	}
+
+	nodeMemoryUsage := nodeMetric.MemoryUsed.MemoryWithoutCache.Value() * 100 / memoryCapacity
+	if nodeMemoryUsage < *thresholdPercent {
+		klog.Warningf("skip memory evict, node memory usage(%v) is below threshold(%v)", nodeMemoryUsage, thresholdConfig)
```
Comment (hormes) [inline pkg/koordlet/resmanager/memory_evict.go:100]: Infof

File: pkg/koordlet/resmanager/memory_evict.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+			bePodInfos = append(bePodInfos, info)
+		}
+	}
+
+	sort.Slice(bePodInfos, func(i, j int) bool {
+		if bePodInfos[i].pod.Spec.Priority == nil || bePodInfos[j].pod.Spec.Priority == nil ||
+			*bePodInfos[i].pod.Spec.Priority == *bePodInfos[j].pod.Spec.Priority {
+			return bePodInfos[i].podMetric.MemoryUsed.MemoryWithoutCache.Value() > bePodInfos[j].podMetric.MemoryUsed.MemoryWithoutCache.Value()
+		}
+		return *bePodInfos[i].pod.Spec.Priority < *bePodInfos[j].pod.Spec.Priority
```
Comment (hormes) [inline pkg/koordlet/resmanager/memory_evict.go:164]: should consider koordinator priority

Comment (hormes) [reply pkg/koordlet/resmanager/memory_evict.go:164]: If the priority is the same, resource specifications and running time need to be considered. Here is where the expansion strategy is required. Abstracting a sorted interface is convenient for expansion?

Comment (hormes) [reply pkg/koordlet/resmanager/memory_evict.go:164]: leave a todo here

File: pkg/koordlet/resmanager/resmanager.go (hunk 1 above)
Comment (jasonliu747) [reply pkg/koordlet/resmanager/resmanager.go:274]: I'll submit another PR later to refine all functions in util package.

Comment (zwzhang0107) [reply pkg/koordlet/resmanager/resmanager.go:274]: remember to add an issue, tks

File: pkg/koordlet/resmanager/memory_evict.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+	}
+
+	klog.Infof("node(%v) MemoryUsage(%v): %.2f, evictThresholdUsage: %.2f",
+		m.resManager.nodeName,
+		nodeMetric.MemoryUsed.MemoryWithoutCache.Value(),
+		float64(nodeMemoryUsage)/100,
+		float64(*thresholdPercent)/100,
+	)
+
+	lowPercent := *thresholdPercent - memoryReleaseBufferPercent
```
Comment (shinytang6) [inline pkg/koordlet/resmanager/memory_evict.go:111]: Why fixed here, not using `MemoryEvictLowerPercent `?

**Pull request merged**